### ISSUE TITLE
Feature Request: disable rules for whole file

### ISSUE THREAD

**Issue by chwo** [realm/SwiftLint#5052]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New feature request

After the introduction of the `blanket_disable_command` rule in https://github.com/realm/SwiftLint/pull/4731, all disabled rules have to be re-enabled again, otherwise the rule produces a warning.

There are some use cases where a rule should not be active for a certain file, e.g. in a `Package.swift` file or test classes.

At the moment this makes it necessary to `disable` the rule at the top of the file and to `enable` it again at the end of the file.

The re-enabling at the end of the file just to avoid the `blanket_disable_command` warning seems cumbersome.

It would be nice to introduce a new `:file` specifier (analog to the existing `:previous`, `:this` or `:next` specifiers) to explicitly convey the intention, that a rule is disabled for the whole file and doesn't need to be re-enabled again.

I think this could also be used to partially solve some uses cases of https://github.com/realm/SwiftLint/issues/2355 

```bash
// swiftlint:disable:file <rule1> [<rule2> <rule3>...]
```

**Comment by SimplyDanny** [comment]
Where is this command allowed in a file? Can it stand basically in any line? Does it then still have an effect on the whole file?

@mildm8nnered: You might be interested in this. I'd also like to hear what you think about the proposal.

**Comment by mildm8nnered** [comment]
> Where is this command allowed in a file? Can it stand basically in any line? Does it then still have an effect on the whole file?
> 
> @mildm8nnered: You might be interested in this. I'd also like to hear what you think about the proposal.

So I think if this was positional (applies to all subsequent lines), then it's basically equivalent to

```
// swiftlint:disable blanket_disable_command
// swiftlint:disable some_other_rule
```

at the top of the file. In my codebase I only have five of those on 1,500 odd files, and SwiftLint has three on ~600, but for some people I could see them having more, and wanting a shortcut.

Having it apply to the whole file from any location in the file makes it a bit too easy to hide away - if that was the case I'd probably SwiftLint against it in my codebase.

**Comment by SimplyDanny** [comment]
> So I think if this was positional (applies to all subsequent lines), then it's basically equivalent to
> 
> ```
> // swiftlint:disable blanket_disable_command
> // swiftlint:disable
> ```
>
> at the top of the file.

Yeah, it'd basically only save the `// swiftlint:disable:next blanket_disable_command` line. The other line is the same. Not so much benefit to be honest.

> Having it apply to the whole file from any location in the file makes it a bit too easy to hide away - if that was the case I'd probably SwiftLint against it in my codebase.

That means introducing this new `file` modifier has to come piggyback with a new rule that triggers on these commands if they appear somewhere not right at the beginning of a file.

**Comment by chwo** [comment]
I would image it to be positional (like the already existing specifiers). So that it applicable downwards from the line it is applied. Making it applicable for the whole file (which means even the lines before it is applied) would be confusion and error-prone in my opinion.

I initially only thought of re-enabling the rule again instead of also disabling the blanket_disable_command at the occurrences instead. So I don't see a major need for this, but I see the benefit of a cleaner interface which conveys the intention more explicitly (which is disabling a rule for this file/from this line downward until the end of the file).

**Comment by SimplyDanny** [comment]
> I would image it to be positional (like the already existing specifiers). So that it applicable downwards from the line it is applied.

Then the modifier `file` is misleading and actually not conveying an explicit intention in my opinion. It says "file" but doesn't mean "file". Instead it's only a part of it.

**Comment by grantneufeld** [comment]
So, perhaps a more precise `swiftlint:disable:rest_of_file`?

**Comment by Jasperav** [comment]
Would be handy for me as well, the blanket_disable_command is triggering while I want to disable rules for a file... annoying
